app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,51 @@
+class OrdersController < ApplicationController
+  before_action :logged_in_user, only: [:new, :create]
+
+  def new
+    @order = Order.new
+    @cart_items = get_line_items_in_cart
+  end
+
+  def create
+    @order = current_user.orders.new
+    @cart_items = get_line_items_in_cart
+    create_order_items
+    @order.assign_attributes(order_params)
+    if @order.save
+      change_product_quantity
```

Comment:
chỗ này vì có 2 thao tác thay đổi DB -> dùng `transaction in rails ` nè. Nhớ tmf hiểu các method phục vụ cho việc rollback data nếu có errors nhé

ngoài ra, suggest chỗ này là dùng callback 
KHi create order item -> thì sẽ thhực hiện gọi `update quantity của product thuộc order item`
tự động luôn, em khỏi handle chỗ này. -> `after create` ở model` order item`
ko clear hú thì hú chị meeting

Reply:
vâng chị em hiểu rồi ạ